### PR TITLE
Switch to `udx-native`

### PR DESCRIPTION
**Pull request by kasperisager**
Supersedes #82. Depends on mafintosh/dht-rpc#40.

### CODE REVIEW

File: lib/connect.js [hunk 1]
```diff
@@ -27,29 +24,45 @@ module.exports = function connect (dht, publicKey, opts = {}) {
     handshake: (opts.createHandshake || defaultCreateHandshake)(keyPair, publicKey),
     request: null,
     requesting: false,
-    protocols: PROTOCOL.TCP | PROTOCOL.UTP,
+    protocols: PROTOCOL.UDX,
```
Comment (kasperisager) [inline lib/connect.js:27]: @mafintosh Do we still need this?

File: lib/socket-pool.js [hunk 2]
```diff
@@ -29,6 +29,10 @@ module.exports = class SocketPool {
     this._free.delete(holder)
   }
 
+  _release (holder) {
+    if (this._sockets.has(holder)) this._free.add(holder)
```
Comment (mafintosh) [inline lib/socket-pool.js:33]: shouldn't we close the socket here if idle also? otherwise who closes these

File: lib/connect.js (hunk 1 above)
Comment (mafintosh) [reply lib/connect.js:28]: no, but let's save this for a "handshake bikeshed session 2022 edition"

**Comment by mafintosh** [comment]
Landed in master, with the additional handshake tweak from our other PR \o/ - leaving this branch around as we have some tests using it

**Pull request closed without merging**